### ISSUE TITLE
MySQL connection error

### ISSUE THREAD

**Issue by nw-bongheelee** [slingdata-io/sling-cli#53]
If the password for the mysql user account contains special characters, an Access denied error occurs.

In my case, if the password is 'password!', an Access denied error occurs, but if I change the password to 'password', I can connect without error.

Can you help me?

**Comment by flarco** [comment]
Hi, I am guessing you have providing the connection in the URL format?
Can you instead provide the key/values in the `env.yaml` file? Sling should encode the password correctly.
See [here](https://docs.slingdata.io/connections/database-connections/mysql#sling-env-file-yaml) for more details.

Should look something like this:
```yaml
connections:
  MYSQL:
    type: mysql
    host: <host>
    password: <password>
    port: <port>
    database: <database>
```

**Comment by nw-bongheelee** [comment]
First of all, thanks for your reply.

However, I have already listed the devices to connect via env.yaml.
The source debits are mysql, postgresql, redshift, bigquery and only mysql is giving me the error.

I also tried wrapping the password in single or double quotes as shown below, and I also tested it with the JDBC connection string.

Additionally, I have tested on MAC, LINUX (CENTOS).

Thanks.
